jruby_executable/src/bin/jruby_build.rs

```diff
@@ -99,8 +114,61 @@ fn jruby_build(args: &Args) -> Result<(), Box<dyn Error>> {
         tar_dir_to_file(&jruby_dir, &tar_file)?;
         bullet = timer.done();
 
-        for arch in &["amd64", "arm64"] {
-            let dir = volume_output_dir.join(base_image.to_string()).join(arch);
+        let tar_path = tar_file.path();
+        let sha = sha256_from_path(tar_path)?;
+        let sha_seven = sha.chars().take(7).collect::<String>();
+        let sha_seven_path = append_filename_with(tar_path, &format!("-{sha_seven}"), ".tgz")?;
+
+        bullet = bullet.sub_bullet(format!("Write {}", sha_seven_path.display(),));
+        fs_err::copy(tar_file.path(), &sha_seven_path)?;
+
+        let timestamp = chrono::Utc::now();
+        for cpu_arch in &[CpuArch::new("amd64")?, CpuArch::new("arm64")?] {
```

Comment:
Have you considered using `inventory::artifact::Arch` here and [here](https://github.com/heroku/docker-heroku-ruby-builder/pull/51/files#diff-1e3d9cad7329371a0f1f11fe6319513f49ce9f9cbe5f8b4d49cbe96752fe01d9R168) instead?

I think that should work "out of the box", while avoiding initializing CpuArch with strings in multiple places, as well as (eliminate the need for [this type conversion implementation](https://github.com/heroku/docker-heroku-ruby-builder/pull/51/files#diff-293260e5637d839b13a4f4a3d15a973ee4e52691440bf6db1dd1855e9bf59ec2R100-R106)).

I think it'd also be worth considering removing `CpuArch` altogether, but that might be a larger refactoring - not sure if it'd be better to do that in a later PR, but at least for this PR it'd probable make more sense to use the types available in `inventory`?

Reply:
I mean something like this :)

```suggestion
        for cpu_arch in [Arch::Amd64, Arch::Arm64] {
```

Reply:
I think it probably would be preferable to remove CpuArch with this change. I pushed a branch with a commit doing that, feel free to cherry-pick that or push that branch to your own. Or delete the branch from this repo if you want

https://github.com/heroku/docker-heroku-ruby-builder/commit/08efc6f4ae7d254f64a099d2200be133f21522ed

Reply:
Awesome suggestion. I pulled your commit into this branch.

Reply:
I think that resolves this conversation, closing. CI is now green.